### PR TITLE
feat: fix for Radio Button

### PR DESCRIPTION
**Pull request by rayyan224**

### CODE REVIEW

File: src/components/Radios/Radios.tsx [hunk 1]
```diff
@@ -22,13 +22,20 @@ const Radios: React.FC<RadiosProps> = ({
...
 }) => {
-    const formattedID = id.replace(/\s/g, '-');
+    const formattedID = id?.replace(/\s/g, '-');
     const isCreditCards = Boolean(typeof items[0] === 'object');
     const [whichIsChecked, setChecked] = useState<number>(
         setWhichIsChecked || items.length,
     );
 
+    useEffect(() => {
+        if (typeof selectedState != 'undefined') {
```
Comment (Y0moo) [inline src/components/Radios/Radios.tsx:34]: Nice solution. This helps to avoid problems with the `0` number

**Comment by Y0moo** [comment]
Why do we need this prop? We have `setWhichIsChecked`, doesn't it cover this functionality?

**Review by Y0moo — CHANGES_REQUESTED** [review]
Why do we need this prop? We have setWhichIsChecked, doesn't it cover this functionality?

File: src/components/Radios/Radios.tsx (hunk 1 above)
Comment (Y0moo) [reply src/components/Radios/Radios.tsx:34]: Since we already have `setWhichIsChecked` we need to replace `selectedState` to `setWhichIsChecked` here

Comment (rayyan224) [reply src/components/Radios/Radios.tsx:34]: Discussed on Slack, this is a breaking a change, merging current solution

**Review by Y0moo — APPROVED** [review]

**Pull request merged**